packages/twenty-server/src/engine/metadata-modules/object-metadata/hooks/before-update-one-object.hook.ts

```diff
@@ -61,9 +62,51 @@ export class BeforeUpdateOneObject<T extends UpdateObjectPayload>
         };
       }
 
-      throw new BadRequestException(
-        'Only isActive field can be updated for standard objects',
-      );
+      if (
+        Object.keys(instance.update).every((key) =>
+          ['labelSingular', 'labelPlural', 'isLabelSyncedWithName'].includes(
+            key,
+          ),
+        )
+      ) {
+        if (
+          Object.prototype.hasOwnProperty.call(
+            instance.update,
+            'isLabelSyncedWithName',
+          ) &&
+          instance.update.isLabelSyncedWithName !== undefined
+        ) {
+          if (
+            instance.update.isLabelSyncedWithName === true &&
+            objectMetadata.standardId
+          ) {
+            const standardObjectMetadata =
+              getStandardObjectMetadataFromDefinition(
```

Comment:
TBH, I would load the metadata from the cache here and filter on standards (actually we only need to filter with standardId)